numpy/f2py/f2py2e.py

```diff
@@ -141,6 +143,19 @@
   --noopt              Compile without optimization
   --noarch             Compile without arch-dependent optimization
   --debug              Compile with debugging information
+  deps:                <dependency1> <dependency2> ... :
+                       Specify additional dependencies that the module relies on.
+                       List dependencies separated by spaces, and use `:' to
+                       signify the end of the dependency list. Dependencies are
+                       stored in a list for further processing.
+
+                       Example: deps: lapack scalapack :
+                       This will identify "lapack" and "scalapack" as dependencies
+                       and remove them from argv, leaving a dependencies list
+                       containing ["lapack", "scalapack"].
+
+                       Note: The trailing `:' is required to signify the end of
+                       the dependency list. This is only relevant for the meson backend.
```

Comment:
Can you explain why this is needed only for Meson?

Reply:
Oh wait, it is mentioned in the PR description. It's a new feature, not a necessary change. It seems okay to me to add, although at some point using `f2py -c` is a bit questionable for non-trivial things of course. The syntax seems unusual though. Wouldn't it be better to have a regular `--dep` that takes a regular string name? And then if you have multiple, just repeat `--dep` for your next dependencies? That would avoid the odd trailing colon, and this is similar to `--config-settings` for `pip` for example.

Reply:
This was implemented this way in keeping with the existing `skip:` and `only:` commands, and because it seemed easier when directly parsing `sys.argv`. Though its much better to use more modern styles, I'll add an `argparse` pre-parse for `sys.argv` to handle `--dep` :)